core/json/mod.rs

```diff
@@ -44,69 +50,112 @@ pub fn get_json(json_value: &OwnedValue) -> crate::Result<OwnedValue> {
     }
 }
 
+pub fn json_array(values: Vec<&OwnedValue>) -> crate::Result<OwnedValue> {
+    let mut s = String::new();
+    s.push('[');
+
+    for (idx, value) in values.iter().enumerate() {
+        match value {
+            OwnedValue::Blob(_) => crate::bail_constraint_error!("JSON cannot hold BLOB values"),
+            OwnedValue::Text(t) => {
+                if t.subtype == TextSubtype::Json {
+                    s.push_str(&t.value);
+                } else {
+                    match crate::json::to_string(&*t.value) {
+                        Ok(json) => s.push_str(&json),
+                        Err(_) => crate::bail_parse_error!("malformed JSON"),
+                    }
+                }
+            }
+            OwnedValue::Integer(i) => match crate::json::to_string(&i) {
```

Comment:
Do we want to DRY out this piece of code into a function?

```rust
match crate::json::to_string(&i) {
                Ok(json) => s.push_str(&json),
                Err(_) => crate::bail_parse_error!("malformed JSON"),
            }
```

Reply:
nah it's fine unless we start to have many non-colocated places where this has to be used a lot